node/tests/swamp/swamp.go

```diff
@@ -0,0 +1,151 @@
+package swamp
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"testing"
+
+	"github.com/ipfs/go-log/v2"
+	"github.com/libp2p/go-libp2p-core/host"
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/stretchr/testify/require"
+
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/libs/keystore"
+	"github.com/celestiaorg/celestia-node/node"
+	"github.com/celestiaorg/celestia-node/node/p2p"
+)
+
+var blackholeIP6 = net.ParseIP("100::")
+
+// TODO(@Bidon15):
+// in case ConnectPeers from Swamp.Network is utilized to connect nodes between each other
+// then p2p_exchange.go line 55 close() will panic due to being closed from dial_sync.go(go-libp2p-swarm lib)
+// in func DialLock, which is calling ad.decref() that closes the channel
+// _, err = sw.Network.ConnectPeers(bridge.Host.ID(), light.Host.ID())
+// require.NoError(t, err)
+type Swamp struct {
+	t            *testing.T
+	Network      mocknet.Mocknet
+	CoreClient   core.Client
+	BridgeNodes  []*node.Node
+	LightClients []*node.Node
+	trustedHash  string
+}
+
+// Need a shared CoreClient in order to pass trusted hash to Bridge and Light Nodes
+func NewSwamp(t *testing.T) *Swamp {
+	if testing.Verbose() {
+		log.SetDebugLogging()
+	}
+
+	ctx := context.Background()
+	core := core.MockEmbeddedClient()
+	swp := &Swamp{t: t, Network: mocknet.New(ctx), CoreClient: core}
+	swp.trustedHash = swp.getHash(context.TODO())
+
+	return swp
+}
+
+func (s *Swamp) WaitTillHeight(height int64) {
+	require.Greater(s.t, height, int64(0))
+
+	bf := core.NewBlockFetcher(s.CoreClient)
+	blocks, err := bf.SubscribeNewBlockEvent(context.TODO())
+	require.NoError(s.t, err)
+
+	for {
+		block := <-blocks
+		if height == block.Height {
+			fmt.Println(block.Height)
+			break
+		}
+	}
+
+	err = bf.UnsubscribeNewBlockEvent(context.TODO())
+	require.NoError(s.t, err)
+}
+
+// createPeer() is a helper for celestia nodes to initialize
```

Comment:
```suggestion
// createPeer is a helper for celestia nodes to initialize
```

you don't need to add the `()` (this is true for all godocs)